app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -158,10 +164,24 @@
 
     private boolean sessionNavigationProceedingAfterOnResume;
 
+    //REMINDER STUFF
```

Comment:
I think we might want to compartmentalize this code a bit more cleanly somewhere other than this activity, since this code and process isn't really bound to the lifecycle of the CommCareHomeActivity.

The Session Service code actually ends up just being able to live inside of that class, which may be a better place for a lot of these helper objects.